Assignment2-Ancestors/Q2.py

```diff
@@ -0,0 +1,41 @@
+from Q1 import find_ancestors, ancestors_by_keys
+from Tree import BinaryTree, Node
+
+def lowest_common_ancestor(tree, key1, key2):
+    """
+    Returns the lowest common ancestor of  two nodes with given keys. Return None
+    in case of any invalid situation (if at least one of the nodes doesn`t exists).
+    :param tree: Tree
+    :param key1: any
+    :param key2: any
+    :return: Node
+    """
+    ancestors1 = find_ancestors(tree, key1)
+    ancestors2 = find_ancestors(tree, key2)
+    if ancestors1 is None or ancestors2 is None: return None
+    common_anc = last_common_element(ancestors1, ancestors2)
+    return common_anc
+
+def last_common_element(list1, list2):

```

Comment:
Having this as a separate function makes `lowest_common_ancestor` above obvious. Thanks for that!